build.gradle

```diff
@@ -128,6 +128,18 @@ task discoJar(type: com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar)
   exclude 'META-INF/*.RSA'
 }
 
+task gapicJar(type: com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar) {
```

Comment:
I think whether or not we have main class should influence the name of the task. If we create an executable jar that runs `CodeGeneratorTool` the task name should "document" that the JAR runs code generator. Otherwise, it's like saying a task for compiling a C program should be called "ELF binary".

If we don't include the main method and require the user to provide the name of the main class, then I think "fatJar" makes sense.